### DISCUSSION TITLE
Start new session Appium server take long time

### DISCUSSION THREAD

**Discussion by nhatpv** [appium/appium#15721] [Q&A] [answered]
I try start Appium server from Python Client.
It take about **~4 or ~5 minutes** to start a new session Appium Server

Below is coding start new session:

```
        time = datetime.now().strftime("%H:%M:%S")
        print("Start new session Appium server time = " + time)
        try:
            appium_service = AppiumService()
            appium_service.start(args=['--address', '127.0.0.1', 
                                    '-p', '4723', 
                                    '-pa', '/wd/hub'])
            time = datetime.now().strftime("%H:%M:%S")
            print("Appium session start successfully time = " + time)
        except WebDriverException as e:
            print(e)
```
            
This is log was output:

> Start new session Appium server time = 11:21:48
Appium session start successfully time = 11:25:01

**Question:**
1. Is timing start new session Appium server Normal?
2. Are there any way to start session only first time?
I try use AppiumService is_running() but it's always return False value.
Would anyone help me confirm. Thanks all!

**Comment by jlipps** [comment]
Is it equally slow if you don't use the AppiumService?

**Reply by nhatpv** [reply]
If I don't use the AppiumService, 
I use Appium Desktop application version v1.21.0 to start session server, it was start quickly, only take ~5, 10 seconds.

**Reply by jlipps** [reply]
OK, this could potentially be reported to the Appium Python client library. or maybe @KazuCocoa has some ideas.

**Reply by KazuCocoa** [reply]
I see. Let me take a look later.

**Comment by KazuCocoa** [comment, marked as answer]
I tried the below commands on my local. Then, the appium server started in a few seconds. The appium was `npm -g` installed v1.21.0.

```
>>> appium_service = AppiumService()
>>> DEFAULT_PORT =
>>> appium_service.start(args=['--address', '127.0.0.1', '-p', '4723', '-pa', '/wd/hub'])
<Popen: returncode: None args: ['/Users/kazuaki/.nodebrew/current/bin/node',...>
>>> appium_service.is_running
True
>>> appium_service.stop()
True
>>> appium_service.is_running
False
```

What version of appium did you run?
How about running your command over CLI?

**Reply by nhatpv** [reply]
This is package version I setup
Package              Version
-------------------- ----------
Appium-Python-Client 1.2.0
pip                                21.2.3
selenium                       3.141.0
Python                          3.9.6

I updated again appium through "npm install -g appium"
It was updated to v1.21.0.
appium server was started in a few second. May be I used old version.

The problem was solved. Sorry for trouble to you.
Thank for your help!
